packages/ember-glimmer/lib/utils/process-args.ts

```diff
@@ -1,36 +1,45 @@
+import { assert } from 'ember-debug';
 import { symbol } from 'ember-utils';
 import { MUTABLE_CELL } from 'ember-views';
 import { CapturedNamedArguments } from '@glimmer/runtime';
 import { ARGS } from '../component';
 import { ACTION } from '../helpers/action';
 import { UPDATE } from './references';
 
-// ComponentArgs takes EvaluatedNamedArgs and converts them into the
+// processComponentArgs takes CapturedNamedArguments and converts them into the
 // inputs needed by CurlyComponents (attrs and props, with mutable
 // cells, etc).
-export function processComponentArgs(namedArgs: CapturedNamedArguments) {
+export function processComponentArgs(namedArgs: CapturedNamedArguments, componentName: string) {
   let keys = namedArgs.names;
   let attrs = namedArgs.value();
   let props = Object.create(null);
   let args = Object.create(null);
 
-  props[ARGS] = args;
+  let seen;
+  if (DEBUG) {
+    seen = Object.create(null);
+  }
 
   for (let i = 0; i < keys.length; i++) {
     let name = keys[i];
+
+    assert(
+      `Component attribute \`${name}\` is declared multiple times for \`${componentName.split(':').slice(-1)}\` component.`,
+      seen[name] ? false : seen[name] = true
+    );
+
     let ref = namedArgs.get(name);
     let value = attrs[name];
 
-    if (typeof value === 'function' && value[ACTION]) {
```

Comment:
Why is this branch removed?  Specifically, before these changes when `value === 'function' && value[ACTION]` was set we set the `attrs[name] = value`. After these changes the `attrs[name]` is going to be a `MutableCell`. This branch was intentionally supporting folks doing `this.attrs.someAction()`, and I _believe_ that is now broken after this change.  

We should confirm that tests exists for `this.attrs.foo()` and add them if they are not present...

Reply:
on [line above](https://github.com/emberjs/ember.js/pull/15653/files#diff-069c99d4ffd5ddba5e8f92d6f8fd040dL22)  `let value = attrs[name]; ` I just removed those lines because it is assigning to attrs what is already there :P `attrs[name] = value; `